sources/cassandra/infoschema.go

```diff
@@ -0,0 +1,242 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cassandra
+
+import (
+	"context"
+	"fmt"
+
+	sp "cloud.google.com/go/spanner"
+	cc "github.com/GoogleCloudPlatform/spanner-migration-tool/accessors/clients/cassandra"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/internal"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/profiles"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/schema"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/sources/common"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/spanner/ddl"
+	"github.com/gocql/gocql"
+)
+
+// InfoSchemaImpl is Cassandra specific implementation for InfoSchema
+type InfoSchemaImpl struct {
+	KeyspaceMetadata cc.KeyspaceMetadataInterface
+	SourceProfile    profiles.SourceProfile
+	TargetProfile    profiles.TargetProfile
+}
+
+// GetToDdl implements the common.InfoSchema interface
+func (isi InfoSchemaImpl) GetToDdl() common.ToDdl {
+	return ToDdlImpl{
+		typeMapper: NewCassandraTypeMapper(),
+	}
+}
+
+// GetTableName returns table name
+func (isi InfoSchemaImpl) GetTableName(schema string, tableName string) string {
+	return tableName
+}
+
+// getTableMetadata returns metadata assosiated with specific table
+func (isi InfoSchemaImpl) getTableMetadata(tableName string) (*gocql.TableMetadata, bool) {
+	if isi.KeyspaceMetadata == nil {
+		return nil, false
+	}
+	for _, tm := range isi.KeyspaceMetadata.Tables() {
+		if tm.Name == tableName {
+			return tm, true
+		}
+	}
+	return nil, false
+}
+
+// GetTables return list of tables in selected keyspace
+func (isi InfoSchemaImpl) GetTables() ([]common.SchemaAndName, error) {
+	if isi.KeyspaceMetadata == nil {
+		return nil, fmt.Errorf("keyspace metadata not initialized")
+	}
+
+	tableMetas := isi.KeyspaceMetadata.Tables()
+	if len(tableMetas) == 0 {
+		return nil, fmt.Errorf("no tables found in keyspace '%s'", isi.SourceProfile.Conn.Cassandra.Keyspace)
+	}
+
+	var tables []common.SchemaAndName
+	keyspace := isi.SourceProfile.Conn.Cassandra.Keyspace
+	for _, tableMeta := range tableMetas {
+		tables = append(tables, common.SchemaAndName{Schema: keyspace, Name: tableMeta.Name})
+	}
+	return tables, nil
+}
+
+// getTypeString is a helper function to get collection types as string
+func getTypeString(typeInfo gocql.TypeInfo) string {
+	switch typeInfo.Type() {
+	case gocql.TypeSet:
+		collectionInfo, ok := typeInfo.(gocql.CollectionType)
```

Comment:
would `collectionInfo.String()` help you give:
`map(type, type)` / `list(type)`/`set(type)`?
If yes, we could simplify by not needing the switchcase.